python/cudf/cudf/core/join/join.py

```diff
@@ -310,8 +316,25 @@ def _merge_results(self, left_result: Frame, right_result: Frame) -> Frame:
             else:
                 del right_names[name]
 
-        # Assemble the data columns of the result:
-        data = left_result._data.__class__()
+        # determine if the result has multiindex columns.  The result
+        # of a join has a MultiIndex as its columns if:
+        # - both the `lhs` and `rhs` have a MultiIndex columns
+        # OR
+        # - either one of `lhs` or `rhs` have a MultiIndex columns,
+        #   and the other is empty (i.e., no columns)
```

Comment:
Dont we also get a multiindex if `left_index=True` and `right_on` is a list of the same number of columns, or the other way around?

Reply:
This comment is not about MultiIndex `index`,  but rather MultiIndex `columns`. I wonder if there's other language we can use to make the distinction more clear? We use `MultiColumn` informally but that term would be unfamiliar to newcomers.

Reply:
Oh I see, thanks for clarifying 👍